### PR TITLE
Shopifolk can use node v20.11.1 for development

### PR DESCRIPTION
**Pull request by jesstelford**
Use the latest version of node for local development. Closes https://github.com/Shopify/polaris-internal/issues/1525

_NOTE_: This _does not_ increase the minimum required node version for consumers of our library. See https://github.com/Shopify/polaris-internal/issues/1524 for that.

I confirmed this does not impact the output of our build processes at all by running a build with node v18.12.0, then again with node v20.11.1, then diffing the generated files (there was no difference as expected).

## Benefits

- Perf: Every version of node gets faster improving DX & build times
- Features: Loads of great new features we can take advantage of
- Allows updating dev deps: Eg, Storybook v8 requires >= Node 18, etc

## Performance

**tl;dr:** Basically no change.

Run using [hyperfine](https://github.com/sharkdp/hyperfine):

```
hyperfine --parameter-list branch node-20-for-devs,main --setup='git checkout {branch} && rm -rf node_modules */node_modules\(N\) && \. ~/.nvm/nvm.sh && nvm use && yarn --prefer-offline' --prepare='rm -rf polaris-for-vscode/dist polaris-icons/dist polaris-migrator/dist polaris-react/build polaris-react/build-internal polaris-tokens/dist polaris.shopify.com/.next' --warmup 1 'yarn build'
```

```
Benchmark 1: yarn build (branch = node-20-for-devs)
  Time (mean ± σ):     12.264 s ±  2.257 s    [User: 0.838 s, System: 3.005 s]
  Range (min … max):    7.118 s … 16.111 s    10 runs

Benchmark 2: yarn build (branch = main)
  Time (mean ± σ):     12.210 s ±  1.932 s    [User: 0.831 s, System: 2.957 s]
  Range (min … max):    8.104 s … 13.980 s    10 runs

Summary
  yarn build (branch = main) ran
    1.00 ± 0.24 times faster than yarn build (branch = node-20-for-devs)
```

### CODE REVIEW

File: .github/workflows/ci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+name: CI
```
Comment (jesstelford) [inline .github/workflows/ci-lint.yml:1]: Decoupling the running of our lint tools (which can be run in any node version) from the running of our tests (which should only be run against the versions of node we support in `package.json#engines`). See https://github.com/Shopify/polaris-internal/issues/1522 for the differences.

File: polaris-tokens/package.json [hunk 2]
```diff
@@ -26,7 +26,6 @@
     "dev": "rollup -c -w",
     "lint": "run-p lint:*",
     "lint:js": "TIMING=1 eslint --cache .",
-    "lint:styles": "stylelint '**/*.{css,scss}'",
```
Comment (jesstelford) [inline polaris-tokens/package.json:29]: This was failing with the error:

```
Error: No files matching the pattern "**/*.{css,scss}" were found.
```

Which makes sense because there's no `.css` or `.scss` files in this folder.

But why was it working before?

Because the previous CI task would (unnecessarily) run `yarn build` which generates `dist/**/*.css` files. So, when the glob was run, there _are_ files for `stylelint` to match against, and hence no error. _But_ those files are all in `dist/` which [is ignored in `stylelintrc.js`](https://github.com/Shopify/polaris/blob/9c24a465c5e001e148bde335bf6319e924f4b1d6/.stylelintrc.js#L43), so `stylelint` would silently ignore all the files and exit successfully having done nothing.

**Review by gwyneplaine — APPROVED** [review]
